localstack-core/localstack/utils/analytics/metadata.py

```diff
@@ -60,6 +62,8 @@ def read_client_metadata() -> ClientMetadata:
         is_ci=os.getenv("CI") is not None,
         is_docker=config.is_in_docker,
         is_testing=config.is_local_test_mode(),
+        product="aws",
```

Comment:
@ackdav As briefly discussed, perhaps we can add an environment variable override here, to allow specialized images (separate products like Azure, Snowflake) to define custom product strings.

Something like (simplified):
```suggestion
        product=os.getenv("LOCALSTACK_TELEMETRY_PRODUCT") or "aws",
```
Same for the edition, and also use the existing `get_localstack_edition()` function as the default.